common/lib/xmodule/xmodule/modulestore/split_mongo/mongo_connection.py

```diff
@@ -203,6 +209,40 @@ def structure_to_mongo(structure, course_context=None):
         return new_structure
 
 
+class CourseStructureCache(object):
+    """
+    Wrapper around django cache object to cache course structure objects.
+    The course structures are pickled and compressed when cached.
+    """
+    def __init__(self):
+        try:
+            self.cache = get_cache('course_structure_cache')
+        except InvalidCacheBackendError:
+            self.cache = get_cache('default')
```

Comment:
I worry about this, because we're not setting any expiration times on these really large documents. Will the regular cache quickly fill up and cause other things we want cached a little longer to be evicted?

Agreed it won't be an issue in our case, but I worry about other installations